x-pack/plugins/observability_solution/inventory/public/components/entity_icon/index.tsx

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import { AGENT_NAME, CLOUD_PROVIDER } from '@kbn/observability-shared-plugin/common';
+import { type CloudProvider, CloudProviderIcon, AgentIcon } from '@kbn/custom-icons';
+import { EuiFlexGroup, EuiFlexItem, EuiIcon } from '@elastic/eui';
+import type { AgentName } from '@kbn/elastic-agent-utils';
+import { euiThemeVars } from '@kbn/ui-theme';
+import { ENTITY_TYPE } from '../../../common/es_fields/entities';
+import type { Entity } from '../../../common/entities';
+
+interface EntityIconProps {
+  entity: Entity;
+}
+
+export function EntityIcon({ entity }: EntityIconProps) {
+  const entityType = entity[ENTITY_TYPE];
+  const defaultSizeIconPx = euiThemeVars.euiSizeL;
+
+  if (entityType === 'host' || entityType === 'container') {
+    const cloudProvider = entity[CLOUD_PROVIDER] as CloudProvider;
+
+    const formattedCloudProvider =
+      typeof cloudProvider === 'string' ? cloudProvider : cloudProvider?.[0];
+
+    return (
+      <EuiFlexGroup
+        style={{
+          width: defaultSizeIconPx,
+          height: defaultSizeIconPx,
+        }}
+        alignItems="center"
+        justifyContent="center"
+      >
+        <EuiFlexItem grow={false}>
+          <CloudProviderIcon
+            cloudProvider={formattedCloudProvider}
+            size="m"
+            title={formattedCloudProvider}
+            role="presentation"
+          />
+        </EuiFlexItem>
+      </EuiFlexGroup>
+    );
+  }
+
+  if (entityType === 'service') {
+    const agentName = entity[AGENT_NAME] as AgentName;
+    const formattedAgentName = typeof agentName === 'string' ? agentName : agentName?.[0];
+
+    return <AgentIcon agentName={formattedAgentName} role="presentation" />;
+  }
+
+  return <EuiIcon type="" size="l" />;
```

Comment:
Should we just return null here?

```suggestion
  return null;
```

Reply:
We can keep it, but please add a comment explaining it.